project-keeper/src/test/java/com/exasol/projectkeeper/validators/files/GitHubWorkflowIOTest.java

```diff
@@ -0,0 +1,152 @@
+                          fetch-depth: 0
+                      - name: Set up JDKs
+                        id: setup-jdks
+                        uses: actions/setup-java@v4
+                        with:
+                          distribution: "temurin"
+                          java-version: |
+                            11
+                            17
+                          cache: "maven"
+                """);
+        final Job job = workflow.getJob("build");
+        assertAll(() -> assertThat(workflow.getOnTrigger(), hasEntry("pull_request", null)),
+                () -> assertThat(job.getRunnerOS(), equalTo("ubuntu-latest")),
+                () -> assertThat(job.getConcurrency(), hasEntry("group", "${{ github.workflow }}-${{ github.ref }}")),
+                () -> assertThat(job.getConcurrency(), hasEntry("cancel-in-progress", true)),
+                () -> assertThat(job.getSteps(), hasSize(2)),
+                () -> assertThat(job.getStep("setup-jdks").getWith(), allOf(hasEntry("distribution", "temurin"), //
+                        hasEntry("java-version", "11\n17\n"), //
+                        hasEntry("cache", "maven"))),
+                () -> assertThat(job.getStep("setup-jdks").getUsesAction(), equalTo("actions/setup-java@v4")));
+    }
+
+    @Test
+    void dumpWorkflow() {
+        final String yaml = testee().dumpWorkflow(GitHubWorkflow
+                .create(Map.of("jobs", Map.of("build", Map.of("steps", List.of(Map.of("id", "step1")))))));
+        assertThat(yaml, equalTo("""
+                jobs:
+                  build:
+                    steps:
+                      - {
+                        id: step1
+                      }
+                """));
+    }
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource("testCases")
+    void dumpLoad(final String testDescription, final String inputYaml, final String expectedFormattedYaml) {
+        final GitHubWorkflowIO yamlIO = testee();
+        final String dumped = yamlIO.dump(yamlIO.load(inputYaml));
+        assertThat(testDescription, dumped, equalTo(expectedFormattedYaml));
+    }
+
+    static Stream<Arguments> testCases() {
+        return Stream.of( //
+                testCase("key-value", "key: value", "{\n  key: value\n}\n"), //
+                testCase("comments removed", "key: value # comment", "{\n  key: value\n}\n"), //
+                testCase("'on' key not converted to true", "on: value", "{\n  on: value\n}\n"), //
+                testCase("'on' value not converted to true", "key: on", "{\n  key: on\n}\n"), //
+                testCase("multi-line list", """
+                        list:
+                          - itemA
+                          - itemB
+                        """, "list: [\n  itemA,\n  itemB\n]\n"),
+                testCase("single-line list", "list: [itemA, itemB]", "list: [\n  itemA,\n  itemB\n]\n"), //
+                testCase("single-line list with quotes", "list: ['itemA', \"itemB\"]",
```

Comment:
Please add test with a list containing items with spaces (where quotes could be required)